github/compliance/issue.py

```diff
@@ -39,43 +41,63 @@ def _issue_labels(
     return frozenset((l.name for l in issue.labels()))
 
 
-def artifact_digest_label(
-    component: cm.Component,
-    artifact: cm.Artifact | str,
-    length: int=18,
-):
+def digest_label(
+    target: cnudie.iter.ResourceNode | cnudie.iter.SourceNode,
+    issue_type: str,
+    max_length: int=50,
+) -> str:
     '''
-    calculates and returns a digest for the given component/resource
+    calculates and returns a digest for the given target and issue_type.
 
     this is useful as GitHub labels are limited to 50 characters
     '''
-    if isinstance(artifact, cm.Resource):
-        name = artifact.name
-    else:
-        name = artifact
 
-    label_str = f'{component.name}:{name}'
+    name = target_name(target=target)
+    prefix = f'{issue_type_digest_label_prefix(issue_type)}/'
+
+    digest_length = max_length - len(prefix)
+    digest_length = int(digest_length / 2) # hexdigest is of double length
 
     # pylint does not know `length` parameter (it is even required, though!)
-    label_dig =  hashlib.shake_128(label_str.encode('utf-8')).hexdigest(length=length) # noqa
+    digest =  hashlib.shake_128(name.encode('utf-8')).hexdigest(
+        length=digest_length,
+    )
 
-    label = f'ocm/resource/{label_dig}'
+    label = f'{prefix}{digest}'
 
-    if len(label) > 50:
-        raise ValueError(f'{length=} would result in label exceeding 50 characters')
+    if len(label) > max_length:
+        raise ValueError(
+            f'{target=} and {issue_type=} would result in label exceeding length of {max_length=}'
+        )
 
     return label
 
 
-def _target_name(
-    target: cnudie.iter.SourceNode | cnudie.iter.ResourceNode,
-):
-    if isinstance(target, cnudie.iter.SourceNode):
-        return f'{target.component.name}:{target.source.name}'
-    elif isinstance(target, cnudie.iter.ResourceNode):
-        return f'{target.component.name}:{target.resource.name}'
+def target_name(
+    target: cnudie.iter.ResourceNode | cnudie.iter.SourceNode,
+) -> str:
+    if isinstance(target, (cnudie.iter.ResourceNode, cnudie.iter.SourceNode)):
+        artifact = cnudie.iter.artifact_from_node(target)
+        return f'{target.component.name}:{artifact.name}'
+
     else:
-        raise NotImplementedError(target)
+        raise TypeError(f'{type(target)} not a valid target')
+
+
+def issue_type_digest_label_prefix(
```

Comment:
some thoughts:

- this function should be marked as "private" (or rather be inlined..)
- label-prefix should depend on the _target_, not on the issue-type in `digest_label` function

-> remove this function, and simply determine the appropriate prefix - inline -

Reply:
Its used two times
- in `issue.py` to get label prefix when calculating the digest label
- in `report.py` to find all issues for an `issue_type` (could be target ofc)

I would rather have a separate non-private function than inline this twice.

Reply:
agree w.r.t. not duplicating the function. change to calculate prefix depending on target's type, though.